openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -261,3 +266,67 @@ def post(self, request, usage_key_str):
         except ValueError as err:
             raise ValidationError(detail=str(err))
         return Response(LibraryXBlockOlxSerializer({"olx": new_olx_str}).data)
+
+
+@view_auth_classes()
+class LibraryBlockAssetListView(APIView):
+    """
+    Views to list an existing XBlock's static asset files
+    """
+    @convert_exceptions
+    def get(self, request, usage_key_str):
+        """
+        List the static asset files belonging to this block.
+        """
+        key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        files = api.get_library_block_static_asset_files(key)
+        return Response(LibraryXBlockStaticFilesSerializer({"files": files}).data)
+
+
+@view_auth_classes()
+class LibraryBlockAssetView(APIView):
+    """
+    Views to work with an existing XBlock's static asset files
+    """
+    @convert_exceptions
+    def get(self, request, usage_key_str, file_path):
+        """
+        Get a static asset file belonging to this block.
+        """
+        key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        files = api.get_library_block_static_asset_files(key)
+        for f in files:
+            if f.path == file_path:
+                return Response(LibraryXBlockStaticFileSerializer(f).data)
+        raise NotFound
+
+    @convert_exceptions
+    def put(self, request, usage_key_str, file_path):
+        """
+        Replace a static asset file belonging to this block.
+        """
+        usage_key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        file_wrapper = request.data['content']
```

Comment:
It will be `multipart/form-data;boundary=SOME_BOUNDARY` with the boundary defined by the HTTP client in use. I chose to use the multipart file uploads API instead of putting the data as the body of the request because it tends to be slightly easier to code the clients this way, and because it doesn't require the client to specify the actual content type of the uploaded file. But I could have also just as easily put the file contents as the body.

Reply:
I see. I think whichever we are using we should explicitly check for that. DRF automatically processes various types and they can end up "just working" and cause compatibility issues in the future.

Reply:
OK I added `parser_classes = (MultiPartParser, )` to remove any autodetection/ambiguity.